SocialCareCaseViewerApi/V1/Gateways/ProcessDataGateway.cs

```diff
@@ -240,19 +240,21 @@ public IOrderedEnumerable<CareCaseData> SortData(string sortBy, string orderBy,
 
             static DateTime? GetDateToSortBy(CareCaseData x)
             {
-                DateTime? dt = null;
-
                 if (string.IsNullOrEmpty(x.DateOfEvent))
                 {
                     bool success = DateTime.TryParseExact(x.CaseFormTimestamp, "dd/MM/yyyy H:mm:ss", CultureInfo.InvariantCulture, DateTimeStyles.None, out DateTime timeStamp);
-                    if (success) dt = timeStamp;
+                    if (success) return timeStamp;
                 }
                 else
                 {
                     bool success = DateTime.TryParseExact(x.DateOfEvent, "dd/MM/yyyy", CultureInfo.InvariantCulture, DateTimeStyles.None, out DateTime dateOfEvent);
-                    if (success) dt = dateOfEvent;
+                    if (success) return dateOfEvent;
+
+                    bool successForISOFormat = DateTime.TryParseExact(x.DateOfEvent, "O", CultureInfo.InvariantCulture, DateTimeStyles.None, out DateTime dateOfEventISOFormat);
+                    if (success) return dateOfEventISOFormat;
```

Comment:
Should this be `if (successForISOFormat)`

Reply:
well spotted, thank you! fixed